capa/ida/plugin/form.py

```diff
@@ -673,19 +683,20 @@ def slot_progress_feature_extraction(text):
 
             update_wait_box("loading rules")
 
+            # function should handle exceptions and return False
             if not self.load_capa_rules():
                 return False
 
-            # just generated above
-            assert self.ruleset_cache is not None
-
             if ida_kernwin.user_cancelled():
                 logger.info("User cancelled analysis.")
                 return False
 
             update_wait_box("extracting features")
 
             try:
+                # just generated above
+                assert self.ruleset_cache is not None
+
```

Comment:
moved `assert` statements under `try`/`except` blocks so we can exit gracefully and not lock up IDA